src/shared_gui_components/BuildingComponentDialogCentralWidget.cpp

```diff
@@ -190,16 +217,17 @@ void BuildingComponentDialogCentralWidget::setTid(const std::string& filterType,
     delete comp;
   }
 
-  RemoteBCL remoteBCL;
-  remoteBCL.setTimeOutSeconds(m_timeoutSeconds);
-  std::vector<BCLSearchResult> responses;
-  if (filterType == "components") {
-    responses = remoteBCL.searchComponentLibrary(searchString.toStdString(), tid, pageIdx);
-  } else if (filterType == "measures") {
-    responses = remoteBCL.searchMeasureLibrary(searchString.toStdString(), tid, pageIdx);
```

Comment:
It was too hard to refractor all the methods around this, instead I fetched all the responses from the BCL if it is the first page or the m_allRepsonses is empty (could just do the latter) then I sort them alphabetically